BD-3914 Add details to Message Activity Log (#9006)
braze-inc/braze-docs: _docs/_user_guide/administrative/app_settings/message_activity_log_tab.md

- User aliasing errors
- A/B testing errors

> - SMS/MMS errors

Reviewer: Should we consider adding `Bad user trigger errors` here?

- User aliasing errors
- A/B testing errors
- SMS/MMS errors
- WhatsApp errors

> - Live Activity Errors

Reviewer 2: ```suggestion
- Live Activity errors
```

- The message was blocked by the receiver as spam.
- Service unavailable, Client host [_IP_ADDRESS_] blocked using Spamhaus.

> ## Storage 

Reviewer: nit: I might add language here to indicate that this is a fixed "retention period". Maybe something like `Storage retention period`.

Errors from the last 60 hours are available in the Message Activity Logs. Logs that are more than 60 hours old are cleaned and no longer accessible. 

### Number of error logs stored

The number of saved logs is influenced by several conditions. For example, if a scheduled campaign is sent to thousands of users, we would potentially see a sample of the errors in the Message Activity Log instead of all errors.

Here's an overview of conditions affecting how many logs will be saved:
- A maximum of 20 Connected Content error logs will be saved for the same campaign within one hour.

> - A maximum of 100 error logs of the same error type will be saved within one hour per workspace for the following error types:

Reviewer: I think we should remove `Connected Content errors` from the following list. Connected Content errors appear to be limited just to 20 per campaign per fixed clock hour, with no cap per workspace ([source](https://github.com/Appboy/platform/blob/e7fb7212fd31803f4b0d1bbe8176beabdfabcc72/shared_code/domains/messaging_pipeline/public/message_activity_log/provider.rb#L178-L180)).

Also, I think we should consider adding `Bad user trigger errors` ([source](https://github.com/Appboy/platform/blob/5e3e3f4196c3ecaf975bd44c96fc9232a0fa142d/shared_code/domains/messaging_pipeline/public/message_activity_log/provider.rb#L186-L197)), since we log them for API triggers with bad users [here](https://github.com/Appboy/platform/blob/e7fb7212fd31803f4b0d1bbe8176beabdfabcc72/shared_code/domains/campaigns/public/appboy/external/api/api_triggered_object_base.rb#L182). Should we add this kind of error to other places we mention specific errors in this doc?

And same question - should we specify that this count is per fixed clock hour, and not per rolling 1-hour window?

Author: Thank you, @archwa! I applied your suggested edits. It might be helpful to specify how fixed cloud hours are determined. Can the user determine this, or is it every new hour in the company time zone? (For example, 1 PM, 2 PM...etc.)

Reviewer: Sure thing! Per [Ruby docs](https://ruby-doc.org/3.1.6/Time.html#:~:text=You%20can%20create%20a%20new%20instance%20of%20Time%20with%20Time.new.%20This%20will%20use%20the%20current%20system%20time.%20Time.now%20is%20an%20alias%20for%20this.), `Time.now` uses system time:
> You can create a new instance of [Time](https://ruby-doc.org/3.1.6/Time.html) with [Time.new](https://ruby-doc.org/3.1.6/Time.html#method-c-new). This will use the current system time. [Time.now](https://ruby-doc.org/3.1.6/Time.html#method-c-now) is an alias for this.

The system time is machine-dependent (whatever machine we are running the code on). So, the fixed clock hours cannot be determined by the user, and they are subject to system clock drift and differences across machines (which probably would be insignificant in most cases).

- The message was blocked by the receiver as spam.
- Service unavailable, Client host [_IP_ADDRESS_] blocked using Spamhaus.

## Storage retention period

Errors from the last 60 hours are available in the Message Activity Logs. Logs that are more than 60 hours old are cleaned and no longer accessible. 

### Number of error logs stored

The number of saved logs is influenced by several conditions. For example, if a scheduled campaign is sent to thousands of users, we would potentially see a sample of the errors in the Message Activity Log instead of all errors.

Here's an overview of conditions affecting how many logs will be saved:

> - A maximum of 20 Connected Content error logs will be saved for the same campaign within one fixed clock hour.

Reviewer 2: ```suggestion
- Up to 20 Connected Content error logs will be saved for the same campaign within one fixed clock hour.
```

> - A maximum of 100 error logs of the same error type will be saved within one fixed clock hour per workspace for the following error types:

Reviewer 2: ```suggestion
- Up to 100 error logs of the same error type will be saved within one fixed clock hour per workspace for the following error types:
```
